src/pages/VeSDL/index.tsx

```diff
@@ -4,52 +4,238 @@ import {
+      }))
+      const vesdlBal = await votingEscrowContract?.["balanceOf(address)"](
+        account,
+      )
+      setVeSdlTokenVal(vesdlBal || Zero)
+
+      const prevLockEnd =
+        (await votingEscrowContract?.locked__end(account)) || Zero
+      setLockEnd(
+        !prevLockEnd.isZero() ? new Date(prevLockEnd.toNumber() * 1000) : null,
+      )
+
+      const veSdlToken = await votingEscrowContract?.locked(account)
+      setLockedSDLVal(veSdlToken?.amount || Zero)
+    }
+  }, [account, sdlContract, votingEscrowContract])
+  const resetFormState = () => {
+    setSDLToken((prev) => ({
+      ...prev,
+      sdlTokenInputVal: "",
+    }))
+    setUnlockDate(null)
   }
 
-  const prevUnlockDate: Date = new Date("2022-5-23")
-  const addLockMos = date ? differenceInMonths(date, prevUnlockDate) : null
+  const currentTimestamp = getUnixTime(new Date())
+  const unlockDateOrLockEnd = unlockDate || lockEnd
+  const expireTimestamp =
+    unlockDateOrLockEnd && !isNaN(unlockDateOrLockEnd.valueOf())
+      ? getUnixTime(unlockDateOrLockEnd)
+      : undefined
+  const totalAmount =
+    !sdlTokenValue.isZero() || unlockDate
+      ? sdlTokenValue.add(lockedSDLVal)
+      : Zero
+
+  // Calculate estimated veSdl amount
+  const estLockAmt = expireTimestamp
+    ? totalAmount
+        .mul(
+          BigNumber.from(expireTimestamp).sub(BigNumber.from(currentTimestamp)),
+        )
+        .div(BigNumber.from(MAXTIME))
+    : Zero
+
+  // Calculate penalty Ratio and penalty amount
+  const leftTimeForUnlock = lockEnd && getUnixTime(lockEnd) - currentTimestamp
+  const penaltyAmount = leftTimeForUnlock
+    ? minBigNumber(
+        lockedSDLVal.mul(BigNumber.from(3)).div(BigNumber.from(4)),
+        lockedSDLVal
+          .mul(BigNumber.from(leftTimeForUnlock))
+          .div(BigNumber.from(MAXTIME)),
+      )
+    : Zero
+
+  const handleLock = async () => {
+    const unlockTimeStamp = unlockDate ? getUnixTime(unlockDate) : null
+    const hasLockedSDL = !lockedSDLVal.isZero()
+    const shouldCreateLock =
+      !hasLockedSDL && sdlTokenValue.gt(Zero) && unlockTimeStamp
+    const shouldIncreaseAmoutAndLockEnd =
+      hasLockedSDL && sdlTokenValue.gt(Zero) && unlockTimeStamp
+    const shouldIncreaseAmount =
+      hasLockedSDL && sdlTokenValue.gt(Zero) && !unlockTimeStamp
+    const shouldIncreaseLockEnd =
+      hasLockedSDL && sdlTokenValue.isZero() && unlockTimeStamp
+
+    if (!account || !chainId || !votingEscrowContract?.address) return
+    try {
+      if (sdlTokenValue.gt(Zero)) {
+        await checkAndApproveTokenForTrade(
+          sdlContract,
+          votingEscrowContract.address,
+          account,
+          sdlTokenValue,
+          infiniteApproval,
+        )
+      }
+
+      if (shouldCreateLock && unlockTimeStamp) {
+        const txn = await votingEscrowContract?.create_lock(
+          sdlTokenValue,
+          unlockTimeStamp,
+        )
+        void enqueuePromiseToast(chainId, txn.wait(), "createLock")
```

Comment:
Does toast work fine with void? I didn't know that

Reply:
Because there is no anything returned from `enquePromiseToast`